sdk/tests/conformance2/extensions/webgl-multi-draw-instanced-base-vertex-base-instance.html

```diff
@@ -0,0 +1,717 @@
+              256.0 * (1.0 - x / x_count),
+              256.0 * (1.0 - y / y_count),
+              (!config.isDrawArrays && config.useBaseVertexBuiltin) ? 256.0 * (1.0 - y / y_count) : 255.0,
+              255.0
+            ],
+            msg + ' (' + x + ',' + y + ')', 1.0
+          ));
+        }
+      }
+      wtu.checkCanvasRects(gl, rects);
+    }
+
+    // Draw functions variations
+
+    function drawArraysInstancedBaseInstance() {
+      const countPerDraw = y_count * 6;
+      for (let x = 0; x < x_count; x += 2) {
+        ext.drawArraysInstancedBaseInstanceWEBGL(gl.TRIANGLES, 0, countPerDraw, 2, x);
+      }
+    }
+
+    function multiDrawArraysInstancedBaseInstance() {
+      ext.multiDrawArraysInstancedBaseInstanceWEBGL(gl.TRIANGLES, drawArraysParams.firsts, 0, drawArraysParams.counts, 0, drawArraysParams.instances, 0, drawArraysParams.baseInstances, 0, drawArraysParams.drawCount);
+    }
+
+    function drawElementsInstancedBaseVertexBaseInstance() {
+      const countPerDraw = 6;
+      for (let v = 0; v < y_count; ++v) {
+        for (let x = 0; x < x_count; x += 2) {
+          ext.drawElementsInstancedBaseVertexBaseInstanceWEBGL(gl.TRIANGLES, countPerDraw, gl.UNSIGNED_SHORT, 0, 2, v * 4, x);
+        }
+      }
+    }
+
+    function multiDrawElementsInstancedBaseVertexBaseInstance() {
+      ext.multiDrawElementsInstancedBaseVertexBaseInstanceWEBGL(gl.TRIANGLES, drawElementsParams.counts, 0, gl.UNSIGNED_SHORT, drawElementsParams.offsets, 0, drawElementsParams.instances, 0, drawElementsParams.baseVertices, 0, drawElementsParams.baseInstances, 0, drawElementsParams.drawCount);
+    }
+
+    function checkDraw(config) {
+      const program = wtu.setupProgram(
+        gl,
+        getShaderSource(x_count, y_count, config),
+        !config.useBaseInstanceBuiltin ? ['vPosition'] : ['vPosition', 'vInstanceID']
+      );
+
+      gl.clear(gl.COLOR_BUFFER_BIT | gl.DEPTH_BUFFER_BIT);
+
+      if (config.isDrawArrays) {
+        gl.bindBuffer(gl.ARRAY_BUFFER, nonIndexedVertexBuffer);
+        gl.enableVertexAttribArray(0);
+        gl.vertexAttribPointer(0, 3, gl.FLOAT, false, 0, 0);
+      } else {
+        gl.bindBuffer(gl.ELEMENT_ARRAY_BUFFER, indexBuffer);
+        gl.bindBuffer(gl.ARRAY_BUFFER, vertexBuffer);
+        gl.enableVertexAttribArray(0);
+        gl.vertexAttribPointer(0, 3, gl.FLOAT, false, 0, 0);
+      }
+
+      if (!config.useBaseInstanceBuiltin) {
+        gl.bindBuffer(gl.ARRAY_BUFFER, instanceIDBuffer);
+        gl.enableVertexAttribArray(1);
+        gl.vertexAttribPointer(1, 1, gl.FLOAT, false, 0, 0);
+        gl.vertexAttribDivisor(1, 1);
+      }
+
+      config.drawFunc();
+      wtu.glErrorShouldBe(gl, gl.NO_ERROR, "there should be no errors");
+
+      checkResult(config);
+    }
+
+    checkDraw({
+      drawFunc: multiDraw ? multiDrawArraysInstancedBaseInstance : drawArraysInstancedBaseInstance,
+      isDrawArrays: true,
+      isMultiDraw: multiDraw,
+      useBaseVertexBuiltin: true,
```

Comment:
Do you plan to add tests with useBaseVertexBuiltin set to false? If so, could you add a TODO about that?

Reply:
Now added all 4 combinations of useBaseVertexBuiltin/useBaseInstanceBuiltin